pulsar-client/src/main/java/org/apache/pulsar/client/impl/PatternConsumerUpdateQueue.java

```diff
@@ -0,0 +1,187 @@
+package org.apache.pulsar.client.impl;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * Used to make all tasks that will modify subscriptions will be executed one by one, and skip the unnecessary updating.
+ *
+ * So far, four three scenarios that will modify subscriptions:
+ * 1. When start pattern consumer.
+ * 2. After topic list watcher reconnected, it will call {@link PatternMultiTopicsConsumerImpl#recheckTopicsChange()}.
+ *    this scenario only exists in the version >= 2.11 (both client-version and broker version are >= 2.11).
+ * 3. A scheduled task will call {@link PatternMultiTopicsConsumerImpl#recheckTopicsChange()}, this scenario only
+ *    exists in the version < 2.11.
+ * 4. The topics change events will trigger a
+ *    {@link PatternMultiTopicsConsumerImpl#topicsChangeListener#onTopicsRemoved(Collection)} or
+ *    {@link PatternMultiTopicsConsumerImpl#topicsChangeListener#onTopicsAdded(Collection)}.
+ *
+ * When you are using a release >= 2.11, there are three scenarios: [1, 2, 4].
+ * When you are using a release < 2.11, there is only one scenario: [3] and all the event will run in the same thread.
```

Comment:
What does "using a release< 2.11" mean? does this mean to connect with this client version to an < 2.11 broker?

Reply:
Correct, improved these comments